base/shared/src/main/scala/scalaz/data/Maybe2Syntax.scala

```diff
@@ -0,0 +1,7 @@
+package scalaz
+package data
+
+trait Maybe2Syntax {
+  def just2 [A, B](a: A, b: B): Maybe2Impl.Maybe2[A, B] = Maybe2Impl.fromOption2(Some2(a, b))
+  def empty2[A, B]            : Maybe2Impl.Maybe2[A, B] = None2
```

Comment:
Why are these two syntax?

Also, `Maybe2Impl` is `private[data]`, so it shouldn't be part of a return type of public method.

Reply:
I have no idea :D Maybe was giving me trouble, this must have been left over... Let me put those back where they belong.

Reply:
By convention, we can (should?) have `MaybeFunctions` for all the top-level functions, and `MaybeSyntax` reserved just for extension methods.

Reply:
How about this?
I added `AllFunctions`, which contains these kinds of methods and is mixed into the prelude.
`Syntax` should contain only extension methods, right?